plugins/services/src/js/components/modals/CreateServiceModal.js

```diff
@@ -491,21 +497,45 @@ class CreateServiceModal extends Component {
     this.setState({ isPending: true });
   }
 
+  getExpandAdvancedSettings() {
+    // If we have errors inside the advanced options, expand the options to allow the user to see the errors.
+    const errors = this.criticalFormErrors();
+
+    if (errors.length > 0) {
+      const { serviceSpec } = this.state;
+      const isDockerContainer =
+        Util.findNestedPropertyInObject(serviceSpec, "container.type") ===
+        "DOCKER";
+
+      const shouldExpand = errors.some(
+        error =>
+          error.path.includes("gpus") ||
+          error.path.includes("disk") ||
+          (isDockerContainer &&
+            error.path.includes("container") &&
+            error.path.includes("docker") &&
+            error.path.includes("image"))
+      );
+
+      return shouldExpand;
+    }
+
+    return false;
+  }
```

Comment:
what do you think of 
1. omitting the `error.length > 0`-check (as the result from the following code will be the same and we don't expect performance issues with a one-time-check.
2. using the complete path instead of checking with an `includes`, as that might break in unexpected ways more easily.

i played around a bit and arrived at the following. it does not check for `container.type === "DOCKER"`, as i thought that this is a dependency of the error occurring at all. that might be wrong though.

```typescript
  getExpandAdvancedSettings() {
    // If we have errors inside the advanced options, expand the options to allow the user to see the errors.
    return this.criticalFormErrors().some(
      ({ path }) =>
        isEqual(path, ["gpus"]) ||
        isEqual(path, ["disk"]) ||
        isEqual(path, ["container", "docker", "image"])
    );
  }
```

Reply:
I will try this.

Reply:
We do need the check for Docker, you are right about the rest. Thanks.